diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt

```diff
@@ -437,52 +403,74 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
                 .forEach {
                     searchBinaryExpression(it, binList)
                 }
-        } else {
+        }
+        if (node.elementType == BINARY_EXPRESSION) {
             binList.add(node)
             binList.add(node.treeParent.findChildByType(PREFIX_EXPRESSION) ?: return)
         }
     }
 
     /**
-     * Collect by Depth-first search (DFS) all children to the right side of the equal sign with specific type [propertyList],
-     * by which we can split expression.
-     * Such logic needed, because AST representation of complex conditions is quite loaded
-     *
-     * @param node target node to be processed
-     * @param binList where to store the corresponding results
+     * Return List of the Pair <node, offset>
+     * First elem in List - Logic Binary Expression (&&  ||)
+     * Second elem in List - Comparison Binary Expression (> < == >= <= !=)
+     * Other types (Arithmetical and Bit operation) (+ - * / % >> << *= += -= /= %= ++ -- ! in !in etc)
      */
-    private fun dfsForProperty(node: ASTNode, binList: MutableList<ASTNode>) {
-        node.getChildren(null).forEach {
-            if (it.elementType in propertyList) {
-                if (it.elementType == REFERENCE_EXPRESSION && it.treeParent?.elementType == CALL_EXPRESSION) {
-                    binList.add(it.treeParent)
-                } else {
-                    binList.add(it)
-                }
-            }
-            dfsForProperty(it, binList)
+    @Suppress("TYPE_ALIAS", "UnsafeCallOnNullableType")
+    private fun searchSomeSplitInBinaryExpression(parent: ASTNode, configuration: LineLengthConfiguration): List<Pair<ASTNode, Int>?> {
+        val logicListOperationReference = listOf(OROR, ANDAND)
+        val compressionListOperationReference = listOf(GT, LT, EQEQ, GTEQ, LTEQ, EXCLEQ)
+        val binList: MutableList<ASTNode> = mutableListOf()
+        searchBinaryExpression(parent, binList)
+        val rightBinList = binList.map {
+            it to positionByOffset(it.getFirstChildWithType(OPERATION_REFERENCE)!!.startOffset).second
         }
+            .sortedBy { it.second }
+            .reversed()
+        val returnList: MutableList<Pair<ASTNode, Int>?> = mutableListOf()
+        addInSmartListBinExpression(returnList, rightBinList, logicListOperationReference)
+        addInSmartListBinExpression(returnList, rightBinList, compressionListOperationReference)
+        val expression = rightBinList.firstOrNull { (it, offset) ->
+            val binExpression = it.getFirstChildWithType(OPERATION_REFERENCE)!!.firstChildNode.elementType
```

Comment:
That's not an binaryExpression now

```suggestion
            val binaryOperationReference = it.getFirstChildWithType(OPERATION_REFERENCE)!!.firstChildNode.elementType
```

Reply:
Done